src/front/glsl_new/ast.rs

```diff
@@ -25,11 +29,35 @@ impl Program {
             types: Arena::<Type>::new(),
             constants: Arena::<Constant>::new(),
             global_variables: Arena::<GlobalVariable>::new(),
+            context: None,
         }
     }
+
+    pub fn get_context(&mut self) -> &mut Context {
```

Comment:
why are we lazily creating the context?

Reply:
Hmmm, my thinking was that it needs to be created when ever a grammar rule needs it, as I'm not entirely sure which order grammar rules are called. But lemme try if can just created it up front.